src/config/plugin.php

```diff
@@ -53,8 +53,8 @@ class Plugin implements Integration {
 	 */
 	public function __construct( Dependency_Management $dependency_management = null, Database_Migration $database_migration = null ) {
 		// @codingStandardsIgnoreStart
-		$this->dependency_management = $dependency_management ?: new Dependency_Management();
```

Comment:
This will break the functionality. The value will become a boolean if the `$dependency_management` is set with a value.

Reply:
Shoot! I missed the short ternary. Let's leave it for now, though I might come back to it later.